sycl/doc/extensions/USM/USM.adoc

```diff
@@ -143,7 +143,7 @@ public:
     typedef usm_allocator<U, AllocKind, Alignment> other;
   };
 
-  usm_allocator();
+  usm_allocator() = delete;
```

Comment:
Along with this change, there is still a missing constructor:
```cpp
template <typename U>
usm_allocator(const usm_allocator<U, AllocKind, Alignment> &other);
```
Some constructors should also be `noexcept`, at least the copy ones.

Also missing `operator==` and `operator!=` as free functions.

Reply:
Changes to allocator will be dealt with in a future PR.